packages/@aws-cdk/aws-ecs/lib/cluster.ts

```diff
@@ -265,6 +265,48 @@ export class EcsOptimizedAmi implements ec2.IMachineImageSource {
   }
 }
 
+/**
+ * Construct a GPU-based Linux machine image from the latest ECS Optimized AMI published in SSM
+ */
+export class EcsGpuOptimizedAmi implements ec2.IMachineImageSource {
```

Comment:
Thanks for the change. I see a lot of the same code, so can we add a refactor here? All logic except the parameter name is the same, so can you pull that into a base class? The actual parameter name can be a constructor argument (and in fact, I think we should make static factory functions to get the various instances).

You can leave the `EcsOptimizedAmi` class for backwards compatibility, but please do put a `@deprecated` tag on it.

Don't forget to add a unit test.

Reply:
Sure, I will update it. Is there some way to write a test that checks for a CFN stack parameter to start with a string? Since the AMI will constantly be changing as we read the value from SSM we can't check for a hard-coded value

Reply:
Hi @mattmcclean -- a unit test that just checks that this is a specifiable field when calling addCapacity should be enough. See: https://github.com/awslabs/aws-cdk/commit/361e1eb251ad17dc1d36a8352fc39a54b5d936c8

Reply:
Is there anything else I need to do? I see you already added a unit test